linetools/spectra/lsf.py

```diff
@@ -364,17 +364,25 @@ def interpolate_to_wv_array(self,wv_array, kind='Akima'):
             # NT: I tried Akima interpolator from scipy.interpolate
             # and is not robust in extreme situations where the
             # wv_array is large compared to the kernel FWHM.
-            #Let's try linetools.analysis.interp Akima version 
+            # Let's try linetools.analysis.interp Akima version
             lsf_vals = interp_Akima(wv_array_AA,lsf_tab['wv'],lsf_tab['kernel'])
 
+        # make sure the kernel is never negative
+        cond = lsf_vals < 0
+        if np.sum(cond) > 0:
+            warnings.warn('The interpolated kernel has negative values; imposing them to be 0.')
+            if debug:
+                import matplotlib.pyplot as plt
+                plt.plot(wv_array_AA, lsf_vals, 'k-')
+                # import pdb; pdb.set_trace()
+            lsf_vals = np.where(lsf_vals < 0, 0., lsf_vals)
+
```

Comment:
I did check and the resulting interpolation is good as expected. Of course, the quality of the recovered LSF kernel depends on the input `wv_array`, if it is too coarse the output won't be nice. I have therefore added an extra check to make sure the LSF kernel is sampled by at least 10 pixels within `wv_array` (see lines 355-360 above).